### PR TITLE
HW

### PR DESCRIPTION
**Pull request by Rommelua**

### CODE REVIEW

File: src/main/java/com/spring/intro/controller/UserController.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        userService.add(new User("Alisa@gmail.com", "2222"));
+        userService.add(new User("Trump@gmail.com", "3333"));
+        userService.add(new User("Merkel@gmail.com", "4444"));
+        return "Users added";
+    }
+
+    @GetMapping("/{userId}")
+    public UserResponseDto get(@PathVariable Long userId) {
+        User user = userService.getById(userId);
+        return convertUserToDto(user);
```
Comment (andromash) [inline src/main/java/com/spring/intro/controller/UserController.java:34]: you can create user in return statement :)

File: src/main/java/com/spring/intro/controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    @GetMapping()
+    public List<UserResponseDto> getAll() {
+        return userService.listUsers().stream()
+                .map(this::convertUserToDto)
+                .collect(Collectors.toList());
+    }
+
+    private UserResponseDto convertUserToDto(User user) {
+        return new UserResponseDto(user.getId(), user.getEmail());
```
Comment (andromash) [inline src/main/java/com/spring/intro/controller/UserController.java:45]: The checklist says: Don’t pass User in the constructor of UserResponseDto, let’s use setters.

File: src/main/java/com/spring/intro/dao/UserDaoImpl.java [hunk 3]
```diff
@@ -44,4 +44,11 @@ public void add(User user) {
             return query.getResultList();
         }
     }
+
+    @Override
+    public User getById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            return session.load(User.class, id);
```
Comment (andromash) [inline src/main/java/com/spring/intro/dao/UserDaoImpl.java]: will the fields of proxy of this object will be generated?

**Review by andromash — COMMENTED** [review]
Good job!

File: src/main/java/com/spring/intro/controller/UserController.java (hunk 2 above)
Comment (Rommelua) [reply src/main/java/com/spring/intro/controller/UserController.java:45]: To my mind phrase "to pass User to constructor" means something like this:
```
    public UserResponseDto(User user) {
        this.id = user.getId();
        this.email = user.getEmail();
    }
```
I don't pass user to constructor - I pass id and email

Comment (andromash) [reply src/main/java/com/spring/intro/controller/UserController.java:45]: I understand, but the last part says 'let's use setters'

File: src/main/java/com/spring/intro/dao/UserDaoImpl.java (hunk 3 above)
Comment (Rommelua) [reply src/main/java/com/spring/intro/dao/UserDaoImpl.java]: forget to change after experiments

File: src/main/java/com/spring/intro/controller/UserController.java (hunk 1 above)
Comment (Rommelua) [reply src/main/java/com/spring/intro/controller/UserController.java:34]: I know, but I don't like to overload code line. Two lines easier to read.

File: src/main/java/com/spring/intro/controller/UserController.java (hunk 2 above)
Comment (Rommelua) [reply src/main/java/com/spring/intro/controller/UserController.java:45]: I don't see any benefit of using setters here.

File: src/main/java/com/spring/intro/dao/UserDaoImpl.java [hunk 4]
```diff
@@ -44,4 +44,11 @@ public void add(User user) {
             return query.getResultList();
         }
     }
+
+    @Override
+    public User getById(Long id) {
```
Comment (jah701) [inline src/main/java/com/spring/intro/dao/UserDaoImpl.java:49]: I would suggest to return Optional<User> from DB 🤔 
But maybe, as always, you've made some magic in here :D

File: src/main/java/com/spring/intro/controller/UserController.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+import java.util.stream.Collectors;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/user")
+public class UserController {
+    private UserService userService;
```
Comment (Sofasmile) [inline src/main/java/com/spring/intro/controller/UserController.java]: better make this field final

File: src/main/java/com/spring/intro/controller/UserController.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        return "Users added";
+    }
+
+    @GetMapping("/{userId}")
+    public UserResponseDto get(@PathVariable Long userId) {
+        User user = userService.getById(userId);
+        return convertUserToDto(user);
+    }
+
+    @GetMapping()
```
Comment (Sofasmile) [inline src/main/java/com/spring/intro/controller/UserController.java]:
```suggestion
    @GetMapping
```

File: src/main/java/com/spring/intro/controller/UserController.java (hunk 2 above)
Comment (Sofasmile) [inline src/main/java/com/spring/intro/controller/UserController.java:45]: is it ok)

File: src/main/java/com/spring/intro/controller/UserController.java (hunk 6 above)
Comment (Rommelua) [reply src/main/java/com/spring/intro/controller/UserController.java]: fixed

File: src/main/java/com/spring/intro/controller/UserController.java (hunk 5 above)
Comment (Rommelua) [reply src/main/java/com/spring/intro/controller/UserController.java]: done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**